src/plugins/email-smtp/util/getMailConfig.js

```diff
@@ -40,76 +39,11 @@ function getConfigFromMailUrl({ logger, mailUrl }) {
   return config;
 }
 
-/**
- * @summary get email sending config for Nodemailer based on well-known service name
- *   https://github.com/nodemailer/nodemailer-wellknown
- * @param {Boolean} logger Whether to log debug messages
- * @param {String} [password] The service password
- * @param {String} service The service name
- * @param {String} [user] The service user
- * @returns {Object} A mail config object
- */
-function getStandardServiceConfig({ logger, password, service, user }) {
-  Logger.debug(`Using ${service} to send email`);
-
-  // get the config from nodemailer-wellknown
-  const conf = getServiceConfig(service);
-  conf.logger = logger;
-
-  // account for local test providers like Maildev with no auth
-  if (!conf.host) {
-    return conf;
-  }
-
-  // add any credentials to the config
-  if (user && password) {
-    conf.auth = { user, pass: password };
-  }
-
-  return conf;
-}
-
-/**
- * @summary get email sending config for Nodemailer based on custom host and port
- * @param {String} host The mail server host
- * @param {Boolean} logger Whether to log debug messages
- * @param {String} [password] The service password
- * @param {Number} port The port number
- * @param {String} [user] The service user
- * @returns {Object} A mail config object
- */
-function getCustomServiceConfig({ host, logger, password, port, user }) {
-  const conf = {
-    host,
-    port,
-    secure: port === 465,
-    logger
-  };
-
-  // don't enforce checking TLS on localhost
-  if (conf.host === "localhost") {
-    conf.ignoreTLS = true;
-  }
-
-  // add any credentials to the config
-  if (user && password) {
-    conf.auth = { user, pass: password };
-  }
-
-  Logger.debug(`Using ${host} to send email`);
-
-  return conf;
-}
-
 /**
  * @summary get the email sending config for Nodemailer
- * @param {Object} context App context
- * @param {String} shopId The ID of the shop for which to get email sending config
  * @returns {{host: String, port: Number, secure: Boolean, auth: Object, logger: Boolean}} returns a config object
  */
-export default async function getMailConfig(context, shopId) {
-  const { collections } = context;
-  const { Packages } = collections;
+export default async function getMailConfig() {
   const { MAIL_URL, EMAIL_DEBUG } = process.env;
```

Comment:
You should update these to come from an `envalid` config file for this plugin while you're in here. `EMAIL_DEBUG` can default to `false` and `MAIL_URL` can default to `""` to make it optional. Then I would remove it from the `.env.example` file. I'm trying to keep things out of that file if possible since it isn't portable when we move plugins to NPM packages.

Reply:
validation added.